### PR TITLE
Run all tests on Travis

### PR DESCRIPTION
**Pull request by joeltine**

### CODE REVIEW

**Comment by joeltine** [comment]
@jeffbcross Jeff, to silence googlebot can you sign the CLA at https://developers.google.com/open-source/cla/individual

**Comment by concavelenz** [comment]
Joel did you look at all at creating a Karma plugin?

**Comment by joeltine** [comment]
BTW, this is only Chrome right now. Once we get this one out, I'll look into enabling the rest.

File: .travis.yml [hunk 1]
```diff
@@ -3,12 +3,24 @@
 # Use of this source code is governed by the Apache License, Version 2.0.
 # See the LICENSE file for details.
 
+# Will upgrade to new container-based infrastructure
```
Comment (concavelenz) [inline .travis.yml]: "Use container-based infrastructure"

File: .travis.yml [hunk 2]
```diff
@@ -3,12 +3,24 @@
...
 sudo: false
 
+cache:
+  - node_modules
```
Comment (concavelenz) [inline .travis.yml]: What is this?

File: protractor_spec.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
+// TODO(joeltine): Remove promise module when stable node version supports it
+// natively.
+var Promise = require('promise');
```
Comment (concavelenz) [inline protractor_spec.js]: consider using goog.Promise here.

File: .travis.yml [hunk 4]
```diff
@@ -3,12 +3,24 @@
...
+
 install:
   - ./scripts/ci/install_closure_deps.sh
 
 script:
-  - java -jar ../closure-compiler/build/compiler.jar -O ADVANCED --warning_level VERBOSE --jscomp_error='*' --jscomp_off=inferredConstCheck --jscomp_off=extraRequire --jscomp_off=unnecessaryCasts --jscomp_off=deprecated --jscomp_off=lintChecks --js='**.js' --js='!**_test.js' --js='!**_perf.js' --js='!**tester.js' --js='!**promise/testsuiteadapter.js' --js='!**osapi/osapi.js' --js='!**svgpan/svgpan.js' --js='!**alltests.js' --js_output_file=$(mktemp)
-
+  - ./scripts/ci/compile_closure.sh
```
Comment (concavelenz) [inline .travis.yml]: Move this to a different PR.

File: scripts/ci/run_all_tests.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+# Script to install dependencies and bootstrap protractor to run all Closure
+# tests.
+
+HTTP_PORT=8080
+
+# Make sure to kill both servers on exit (e.g., CTRL+C).
+trap 'fuser -k $HTTP_PORT/tcp' EXIT
+
+# Start web server.
+python scripts/http/simple_http_server.py 2> /dev/null & sleep 5
```
Comment (concavelenz) [inline scripts/ci/run_all_tests.sh]: Can you explain why can't use the SimpleHttpServer directly?

File: .travis.yml (hunk 2 above)
Comment (joeltine) [inline .travis.yml]: Tell travis to persist this directory across runs. It holds our required nodejs modules.  Added comment.

File: scripts/ci/run_all_tests.sh (hunk 5 above)
Comment (joeltine) [inline scripts/ci/run_all_tests.sh]: It doesn't support POST.

File: .travis.yml (hunk 1 above)
Comment (joeltine) [inline .travis.yml]: done.
